libs/tools/sharepoint/state.js

```diff
@@ -0,0 +1,45 @@
+import { signal } from '../../deps/htm-preact.js';
+
+const CONFIG = '/.milo/config.json';
+
+// Signals
+export const statuses = signal({});
+export const heading = signal({ name: '' });
+export const languages = signal([]);
+export const urls = signal([]);
+export const siteConfig = signal(null);
+export const user = signal();
+export const spAccessToken = signal();
+
+export function setStatus(name, type, text, description, timeout) {
+  const content = type && text ? { type, text, description } : null;
+  statuses.value = { ...statuses.value, [name]: content };
+  if (timeout) {
+    setTimeout(() => {
+      delete statuses.value[name];
+      statuses.value = { ...statuses.value };
+    }, timeout);
+  }
+}
+
+export function getSiteConfig() {
```

Comment:
Seems like there's a bunch of locui code in here that doesn't need to be in here.